bindata/etcd/cluster-restore.sh

```diff
@@ -66,6 +66,18 @@ function wait_for_containers_to_stop() {
     echo "complete"
   done
 }
+function check_snapshot_status() {
+  dl_etcdctl
+  etcdctl snapshot status "${SNAPSHOT_FILE}" -w json > "${ETCD_DATA_DIR_BACKUP}"/etcd-backup-status-from-snapshot.json
+  tar -C "${ETCD_DATA_DIR_BACKUP}" -xzf "${BACKUP_FILE}" etcd-backup-status.json || true
+  # With the older backups, the backup status file may not exist.
+  if [ -f "${ETCD_DATA_DIR_BACKUP}/etcd-backup-status.json" ]; then
+    if ! diff -q "${ETCD_DATA_DIR_BACKUP}/etcd-backup-status.json" "${ETCD_DATA_DIR_BACKUP}/etcd-backup-status-from-snapshot.json" > /dev/null; then
+       echo "The snapshot file status hash does not match with the hash at the time of the backup. Exiting!"
```

Comment:
can we make the error more clear the user may not understand what snapshot means or which hash as the context of the command is backup/restore. If you are goping to store all of the state from status the error should perhaps reflect which key failed. So maybe we only care about the hash as it precludes the others have changed?

```
backup integrity verification failed SHA256SUM mismatch
```